### ISSUE TITLE
Cleanup Database Fixtures

### ISSUE THREAD

**Issue by mvandeberg** [steemit/steem#1936]
I was writing a reply to a question in #1872 and it ended up being a description for an issue.

> What is the difference between database_fixture and clean_database_fixture?

This is a bit of an artifact from HF 1 which included the reverse vesting share split. In order to accurately test the split, I loaded the database from the live blockchain and tested the hardfork there. Starting the database for unit tests and the live blockchain are different processes (now made a bit similar through appbase), but there was still need for shared behavior.

Any shared helper functions were shoved up to `database_fixture`. Unit test specific startup was moved to a subclass, `clean_database_fixture`, and functions specifically for running a test on data from the main net was moved to a subclass, `live_database_fixture`. The vesting split test can be deleted along with the `live_database_fixture`. It is effectively tested on reindex and the idea behind the `live_database_fixture` is more or less handled via a test net.

We cannot move the `clean_database_fixture` back in to `database_fixture` because there are a few unit tests that do some unique set up and need to handle that themselves rather than use the automatic setup in the constructor of `live_database_fixture`. Alternatively, we can move the initialization out of the constructor altogether and explicitly call it in each test.

**Comment by theoreticalbts** [comment]
So what is the resolution to this ticket?  Does / should #1872 become blocked on this ticket?  Or should we go ahead and merge #1872 ?

**Comment by mvandeberg** [comment]
Let's merge #1872

I do not view this issue as critical. It is a unit test refactor to clean up the code. It is a lower priority. It is in the backlog for when we have a little bit of down time.

Specifically in reference to #1872, if we move setup to a separate method, then the we need a single database_fixture. SMTs can subclass that one if they want to have the SMT code separated. We go from 5 fixtures with a complicated inheritance scheme to 2 with a single inheritance.
